src/gc.c

```diff
@@ -519,17 +577,24 @@ static inline uint16_t gc_setmark_pool_(jl_ptls_t ptls, jl_taggedvalue_t *o,
     uint16_t tag_changed = !gc_marked(tag);
     if (tag_changed) {
         if (mark_mode == GC_OLD_MARKED) {
-            perm_scanned_bytes += page->osize;
-            page->nold++;
+            ptls->gc_cache.perm_scanned_bytes += page->osize;
+            jl_atomic_fetch_add_relaxed(&page->nold, 1);
         }
         else {
-            scanned_bytes += page->osize;
+            ptls->gc_cache.scanned_bytes += page->osize;
+            if (mark_reset_age) {
+                page->has_young = 1;
+                char *page_begin = gc_page_data(o) + GC_PAGE_OFFSET;
+                int obj_id = (((char*)o) - page_begin) / page->osize;
+                uint8_t *ages = page->ages + obj_id / 8;
+                jl_atomic_fetch_and_relaxed(ages, ~(1 << (obj_id % 8)));
```

Comment:
this seems expensive. maybe we should make `ages` one byte-per-object, so this can be `mov` instead of `lock and`? (I realize the cacheline stall could still be pretty bad, but at least that wouldn't always be incurred?)